### ISSUE TITLE
Subsequent lines become indented

### ISSUE THREAD

**Issue by rivenirvana** [kovidgoyal/kitty#5381]
**Describe the bug**
Since commit 97d7329, subsequent lines of every command (even the rendering of the prompt) are indented with a size of the current length of the window, instead of just placing it below in its own line. This bug does not occur when kitty is launched with `--hold`, and occurs whether or not `--config NONE` is set at launch.

**To Reproduce**
1. Launch kitty with no arguments
2. Run any command

**Screenshots**
Observed behavior:
![Screenshot from 2022-08-14 16-03-31](https://user-images.githubusercontent.com/43519644/184528975-5ddfd971-ad96-4cd4-96bb-0a41736feba6.png)
Expected behavior:
![Screenshot from 2022-08-14 16-32-20](https://user-images.githubusercontent.com/43519644/184529011-10da22a6-8424-451a-a916-48bbf6cced06.png)

**Additional context**
Using Fedora 36, I have kitty installed via a [COPR](https://copr.fedorainfracloud.org/coprs/agriffis/kitty-nightly/) that builds daily from master. Comparing the build which introduced the bug with a previous build that does not narrowed down 97d7329 as the commit containing it.
I don't know a lot about kitty's code and development, so I just looked at what the commit changes. I was successful in restoring expected behavior after rebuilding kitty with the following changes:
```
@@ -240,7 +240,7 @@ restore_termios_settings(void) {
         // this is to try to detect if something like less changed the termios state. This allows, for instance kitty @ ls | less to work
         struct termios current_state;
         if (get_termios_state(&current_state) && memcmp(&self_termios, &current_state, sizeof(current_state)) == 0) {
-            safe_tcsetattr(self_ttyfd, TCSAFLUSH, &restore_termios);
+            safe_tcsetattr(self_ttyfd, TCSANOW, &restore_termios);
             termios_needs_restore = false;
         }
     }
@@ -255,7 +255,7 @@ reapply_termios_settings(void) {
 static void
 cleanup(void) {
     child_pid = 0;
-    restore_termios_settings();
+    if (self_ttyfd > -1 && termios_needs_restore) { safe_tcsetattr(self_ttyfd, TCSAFLUSH, &restore_termios); termios_needs_restore = false; }
 #define cfd(fd) if (fd > -1) { safe_close(fd); fd = -1; }
     cfd(child_master_fd); cfd(child_slave_fd);
     cfd(self_ttyfd); cfd(socket_fd); cfd(signal_read_fd); cfd(signal_write_fd);
```
Specifically, changing `TCSAFLUSH` in `restore_termios_settings()` to `TCSANOW`, and reverting the change to `cleanup()`, as they were before the commit.

Note that I do not know if this is the correct approach and if it completely fixes the issue, or if it affects the purpose of the commit as I am unfamiliar with the feature it's trying to amend. I would like to know if anyone else has been experiencing this issue as well in different environment/build setups.

**Comment by kovidgoyal** [comment]
Are you calling some kitty program in your shell's rc files? Most likely in something like PROMPT_COMMAND or similar?

**Comment by rivenirvana** [comment]
Aha, you're right. It's been a while since I've modified it, but I found `kitty + complete setup zsh | source /dev/stdin` in my zshrc to acquire shell completion. I'm pretty sure I found that in the docs, but apparently I don't need it now. Or I never did and the recent commit just emphasized that.

Anyway, there are completion files in the packages now, so removing it just fixes the problem for me. Closing this issue now. Thanks a lot!
